### PR TITLE
Summary use case

### PR DESCRIPTION
**Pull request by jchionak**
Generates summaries for podcasts based on their transcriptions using the ChatGPT API.

### CODE REVIEW

File: src/main/java/use_case/summary/SummaryInteractor.java [hunk 1]
```diff
@@ -13,16 +13,25 @@ public class SummaryInteractor implements SummaryInputBoundary {
...
         Episode episode = inputData.getEpisode();
-        try {
-            String summary = APIWrapper.generateSummary(episode.getTranscript());
-            episode.setSummary(summary);
-            // once the summary is generated, set the summary and save the episode
-            episodeDataAccessObject.saveEpisode(episode);
-            SummaryOutputData outputData = new SummaryOutputData(episode, false);
-            outputBoundary.prepareSuccessView(outputData);
-        } catch (IOException e) {
+        if (episode.getSummary() != null) {
```
Comment (AliShaban1) [inline src/main/java/use_case/summary/SummaryInteractor.java:16]: There might be a small logical error in this if statement, maybe you need episode.getsummary == null?

File: src/main/java/entities/Episode.java [hunk 2]
```diff
@@ -17,7 +17,7 @@ public Transcript getTranscript() {
         return transcript;
     }
 
-    public String getSummary() {
-        return summary;
-    }
+    public String getSummary() {return summary;}
+    public void setSummary(String summary) {this.summary = summary;}
```
Comment (AliShaban1) [inline src/main/java/entities/Episode.java:21]: can you reformat this?

**Review by AliShaban1 — CHANGES_REQUESTED** [review]
Great! just make sure to make a couple of small adjustments

File: src/main/java/use_case/summary/SummaryInteractor.java (hunk 1 above)
Comment (jchionak) [reply src/main/java/use_case/summary/SummaryInteractor.java:16]: You're right, I made this change and forgot to commit it haha

Comment (AliShaban1) [reply src/main/java/use_case/summary/SummaryInteractor.java:16]: haha no worries

**Review by AliShaban1 — APPROVED** [review]
Great job!

**Pull request merged**